src/c++/perf_analyzer/genai-perf/genai_perf/llm_inputs/output_format_converter.py

```diff
@@ -0,0 +1,275 @@
+# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import random
+from typing import Dict, List
+
+from genai_perf.exceptions import GenAIPerfException
+from genai_perf.llm_inputs.shared import (
+    DEFAULT_TENSORRTLLM_MAX_TOKENS,
+    ModelSelectionStrategy,
+    OutputFormat,
+)
+
+
+class OutputFormatConverterFactory:
+    @staticmethod
+    def create(output_format: OutputFormat):
+        converters = {
+            OutputFormat.OPENAI_CHAT_COMPLETIONS: OpenAIChatCompletionsConverter,
+            OutputFormat.OPENAI_COMPLETIONS: OpenAICompletionsConverter,
+            OutputFormat.OPENAI_EMBEDDINGS: OpenAIEmbeddingsConverter,
+            OutputFormat.RANKINGS: RankingsConverter,
+            OutputFormat.VLLM: VLLMConverter,
+            OutputFormat.TENSORRTLLM: TensorRTLLMConverter,
+        }
+        if output_format not in converters:
+            raise GenAIPerfException(f"Output format {output_format} is not supported")
+        return converters[output_format]()
+
+
+class BaseConverter:
+    def convert(
+        self,
+        generic_dataset: Dict,
```

Comment:
A ton of arguments to all these Converters. (I know that I'm responsible for this mess!) It would be cleaner (and more maintainable) if you created an interface `dataclass`  and used this for all these methods.

Reply:
Do we need to specify all these arguments in the `BaseConverter`? The method needs to be overloaded by the derived class anyway so I don't think we need all these arguments in the base method. Why don't we just remove it entirely? or make it
```python
    def convert(self, **kwargs) -> Dict:
        raise NotImplementedError
```

Reply:
We use a config class for this in the exporters. I second this recommendation

Reply:
I think we should _not_ hide the arguments passed to the function inside another dataclass. This hinders readability of the code and add extra layer of complexity to the function. If the function has too many arguments, it means the function is too complex and the solution is to simplify the function, rather than moving the arguments into separate class.

Reply:
I dont think this is an either or situtation.
Simplify and use a dataclass for the arguments if they are a logical grouping.

Reply:
> I dont think this is an either or situtation.
what do you mean?

Also, I think we should just remove this method from the base. Why we need it in the base method if the derived classes all overload the method?

Reply:
I think i was more talking in the general sense of how to move the code to a better cleaner place.
if all of the classes are overloading it, but the base class is trying to create an interface...a NotImplementedError is the correct approach.

Reply:
If we want to enforce having some kind of interface, sure I'm fine with that. I would suggest removing the overloaded parameters from the base method in that case:
```python
    def convert(
        self,
        generic_dataset: Dict,
        add_stream: bool,
        extra_inputs: Dict,
        model_name: list,
        model_selection_strategy: ModelSelectionStrategy,
    ) -> Dict:
        raise NotImplementedError
```

(not sure if `add_stream` should be removed as well).

Reply:
Summary of offline work: these optimizations will be done in a future PR or as part of the modularization work. It has been saved in the "future work" list in the PR description. Any of those items that are not covered will be made into their own tickets before/once the feature branch is merged into main.

Right now, we're going to make as few changes as possible while splitting the LLM Inputs file into multiple files as a step in the direction of modularization. The goal of this feature branch is to make minimal changes while splitting the monolithic file into several files with no behavior change.